jerry-core/api/jerry.c

```diff
@@ -254,6 +255,44 @@ jerry_gc (void)
   ecma_gc_run (JMEM_FREE_UNUSED_MEMORY_SEVERITY_LOW);
 } /* jerry_gc */
 
+#ifdef JMEM_STATS
+void jmem_heap_get_stats (jmem_heap_stats_t *);
+#endif
+/**
+ * Get heap memory stats.
+ *
+ * @return true - get the heap stats successful
+ *         false - otherwise. Usually it is because the MEM_STATS feature is not enabled.
+ */
+bool
+jerry_get_memory_stats (jerry_heap_stats_t *out_stats_p) /**< [out] heap memory stats */
+{
+#ifdef JMEM_STATS
+  if (out_stats_p == NULL)
+  {
+    return false;
+  }
+
+  jmem_heap_stats_t jmem_heap_stats = {0};
+  jmem_heap_get_stats (&jmem_heap_stats);
+
```

Comment:
I don't like the reserved fields. But this is only my personal preference. My experience is that they are always remaining intact. This is not an objection. If everyone vote for reserved fields than I can accept that. 

> But I feel memory statistics (as the name suggests) is different. It would be great to have at least a plan for what will happen then.

I think this won't change frequently. @zherczeg do you have any plan with memory statistics or management in the near future?